### DISCUSSION TITLE
Fixed height gets overwritten after using a search function

### DISCUSSION THREAD

**Discussion by thomasweigert91** [neuefische/questions#373] [Web] [answered]
i have the problem that a scrollbar, which should take a filter function, disappears as soon as i search for elements via a search bar. We have already given the scrollbar a fixed height with !important, but it still gets squeezed every time.

EDIT: The code does not show the map function. Dont know, why

here is the code:
**Horizontal Scrollbar:**
```js
const HorizontalScroll = ({data, bodyPart, setBodyPart}) => {
return (
<HorizontalScrollContainer>
{data.map(item => (
<BodyPart
key={item.id}
item={item}
bodyPart={bodyPart}
setBodyPart={setBodyPart}
/>
))}
</HorizontalScrollContainer>
);
};

const HorizontalScrollContainer = styled.div`
  width: 100vw;

  margin: 1rem auto;
  display: flex;
  padding: 1rem;
  align-items: center;
  overflow-y: hidden;
  margin-bottom: 3rem;
  border: 5px solid yellow;
  height: 100px !important;
`;
```

**This component shows up after searching for different tags:**

```js
const Exercises = ({exercises, setExercises, bodyPart}) => {
return (
<ExercisesContainer>
<ExercisesHeading>See results</ExercisesHeading>
<ExercisesDiv>
{exercises.map((exercise, index) => (
<p key={index}>{exercise.name}</p>
))}
</ExercisesDiv>
</ExercisesContainer>
);
};

const ExercisesContainer = styled.section`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  margin-top: 5rem;
`;

const ExercisesDiv = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

const ExercisesHeading = styled.p``;
```

**Comment by actyvystom** [comment, marked as answer]
Hi Thomas, please use code blocks to display your code correctly, otherwise it's hard to get a picture of what's going wrong there...
Thanks!

**Reply by thomasweigert91** [reply]
I don't know, how it works... last time, i have posted my code, everything worked fine from the beginning

**Reply by actyvystom** [reply]
you can use a block like this: 
<pre>
```js
your code
```
</pre>

**Reply by thomasweigert91** [reply]
Now it works, thanks.

**Reply by actyvystom** [reply]
Just for completeness of the answer:
The surrounding parent container had a height of `100vh`
After changing to `100%` it let all child elements their defined height.
